autotest/gcore/vsihdfs.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+###############################################################################
+# $Id$
+#
+# Project:  GDAL/OGR Test Suite
+# Purpose:  Test VSI file primitives
+# Author:   James McClain <jmcclain@azavea.com>
+#
+###############################################################################
+# Copyright (c) 2011-2013, Even Rouault <even dot rouault at mines-paris dot org>
+# Copyright (c) 2018, Azavea
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+###############################################################################
+
+import sys
+import os
+from osgeo import gdal
+
+sys.path.append('../pymod')
+
+import gdaltest
+
+# Read test
+def vsihdfs_1():
+    filename = '/vsihdfs/file:' + os.getcwd() + '/data/text.txt'
+    fp = gdal.VSIFOpenL(filename, 'rb')
+    if fp is None:
+        return 'skip'
+
```

Comment:
and here gdaltest.have_vsihdfs = True, and then in the following test step you would test this instead of trying to reopen the file when you don't need it

Reply:
Okay, I have done this.